bugbug/model.py

```diff
@@ -221,13 +223,29 @@ class BugModel(Model):
     def get_id(self, bug):
         return bug["id"]
 
-    def items_gen(self):
+    def items_gen(self, classes):
         return (bug for bug in bugzilla.get_bugs())
 
 
 class CommitModel(Model):
     def get_id(self, commit):
         return commit["node"]
 
-    def items_gen(self):
+    def items_gen(self, classes):
         return (commit for commit in repository.get_commits())
+
+
+# for the duplicate model class
+class BugCoupleModel(Model):
+    def get_id(self, bug):
+        return bug[0]["id"], bug[1]["id"]
+
+    def items_gen(self, classes):
+        bugs = {}
+        for bug in bugzilla.get_bugs():
+            bugs[bug["id"]] = bug
+        res = []
```

Comment:
No need to create a big list here that you will throw away rigth after iterating through it.

You can return a generator instead, like the other items_gen:
```
return ((bugs[bug_id1], bugs[bug_id2]) for bug_id1, bug_id2 in classes)
```